PolyEngine/RenderingDevice/OpenGL/Src/GLRenderingDevice.hpp

```diff
@@ -75,6 +76,7 @@ namespace Poly
 		void RenderWireframe(World* world, const AARect& rect, CameraComponent* cameraCmp) const;
 		void RenderNormals(World* world, const AARect& rect, CameraComponent* cameraCmp) const;
 		void RenderNormalsWireframe(World* world, const AARect& rect, CameraComponent* cameraCmp) const;
+		void RenderDebug(World* world, const AARect& rect, CameraComponent* cameraCmp) const;
```

Comment:
I'm pretty sure this should be totaly independent of current rendering mode (we might need debug arrows in lit as well as in unlit and wireframe modes)

Reply:
I agree. How do you see this done? As single call:
```
GeometryRenderingPasses[eGeometryRenderPassType::IMMEDIATE_DEBUG]->Run(world, cameraCmp, rect);
```
in right place in `GLRenderingDevice::RenderUnlit`, `GLRenderingDevice::RenderLit`, etc?

Reply:
I would leave it here for now. Once Tiled Forward and particles are merged we will have better picture of whole rendering process and how to organize opaque, transparent, postprocess and debug passes.

Reply:
So be it. I'm adding proper trask on trello.